### PR TITLE
Forbid use of middleware blacklisting along with whitelisting

### PR DESCRIPTION
**Pull request by iamluc**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | yes
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #194 
| License       | Apache License 2.0

### CODE REVIEW

File: src/Resources/doc/middleware.md [hunk 1]
```diff
@@ -41,7 +41,7 @@ csa_guzzle:
             middleware: ['!my_middleware']
 ```
 
-Disabled middleware have priority over enabled middleware.
+Warning: you cannot use the "disable" feature along with the "enable" feature.
```
Comment (csarrazi) [inline src/Resources/doc/middleware.md]: I would change the wording as "You can either whitelist or blacklist middleware. Using both whitelisting and blacklisting will trigger an exception."

File: src/DependencyInjection/CompilerPass/MiddlewarePass.php [hunk 2]
```diff
@@ -173,8 +174,18 @@ private function filterClientMiddleware(array $middlewareBag, array $tags)
             }
         }
 
-        return array_filter($middlewareBag, function ($value) use ($whiteList, $blackList) {
-            return in_array($value['alias'], $whiteList, true) && !in_array($value['alias'], $blackList, true);
-        });
+        if ($whiteList && $blackList) {
```
Comment (csarrazi) [inline src/DependencyInjection/CompilerPass/MiddlewarePass.php]: Don't forget the `@throws` annotation at the top of the method.

**Comment by iamluc** [comment]
Updated

**Comment by csarrazi** [comment]
Thanks @iamluc!

**Pull request merged**